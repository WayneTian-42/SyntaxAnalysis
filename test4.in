D -> TV
T -> t | f
V -> sR
R -> ,V | #