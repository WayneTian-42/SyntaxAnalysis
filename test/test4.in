E -> TH
H -> +TH | #
T -> FR
R -> *FR | #
F -> (E) | i