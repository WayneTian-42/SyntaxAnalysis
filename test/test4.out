First Set
E  ( i 
F  ( i 
H  # + 
R  # * 
T  ( i 

Follow Set
E  $ ) 
F  $ ) * + 
H  $ ) 
R  $ ) + 
T  $ ) + 

Predict Analysis Table
        (              )              *              +              i              $              
E       E -> TH                                                     E -> TH                       
H                      H -> #                        H -> +TH                      H -> #         
T       T -> FR                                                     T -> FR                       
R                      R -> #         R -> *FR       R -> #                        R -> #         
F       F -> (E)                                                    F -> i                        

Please enter the symbols:
*i*+i

Stack               Input               Output              
$E                  *i*+i$              error detect: point++
$E                  i*+i$               E -> TH             
$HT                 i*+i$               T -> FR             
$HRF                i*+i$               F -> i              
$HRi                i*+i$               accept i
$HR                 *+i$                R -> *FR            
$HRF*               *+i$                accept *
$HRF                +i$                 error detect: pop out F
$HR                 +i$                 R -> #              
$H                  +i$                 H -> +TH            
$HT+                +i$                 accept +
$HT                 i$                  T -> FR             
$HRF                i$                  F -> i              
$HRi                i$                  accept i
$HR                 $                   R -> #              
$H                  $                   H -> #              
$                   $                   accept $
Error detect but accept